core/modules/system/js/modules.js

```diff
@@ -90,10 +90,19 @@ Backdrop.behaviors.moduleFilter = {
       }
     }
 
-    // Clear out the input field when clicking the reset button.
+    // Clear out the input field and search query when clicking the reset
+    // button.
     function resetModuleList(e) {
+      // Clear input field.
       $input.val('').triggerHandler('keyup');
       e.preventDefault();
+
+      // Clear search query.
+      var uri = window.location.toString();
+      if (uri.indexOf("?") > 0) {
+          var clean_uri = uri.substring(0, uri.indexOf("?"));
+          window.history.replaceState({}, document.title, clean_uri);
```

Comment:
```suggestion
        var clean_uri = uri.substring(0, uri.indexOf("?"));
        window.history.replaceState({}, document.title, clean_uri);
```

Reply:
...I believe that we can reduce the entire code here simply by chaining `.split('?')[0]` to the `uri` variable, which will work even when there is no query string in the URL, and which won't require any conditional check:

`var uri = window.location.toString().split('?')[0];`

Reply:
...actually, turns out that we don't need to do any variable declaration and processing at all - we can use native functions and simply do this:

`window.location.href = window.location.origin + window.location.pathname;`

Courtesy of https://stackoverflow.com/a/48542058
- [`location.origin`](https://developer.mozilla.org/en-US/docs/Web/API/Location/origin): ...a string containing the Unicode serialization of the origin of the represented URL. ... That is: ... for URL using the `http` or `https`, the scheme followed by '://', followed by the domain, followed by `':'`, followed by the port (the default port, `80` and `443` respectively, if explicitly specified);
- [`location.pathname`](https://developer.mozilla.org/en-US/docs/Web/API/Location/pathname): ...a string containing the path of the URL for the location. If there is no path, `pathname` will be empty: otherwise, `pathname` contains an initial '/' followed by the path of the URL, not including the query string or fragment.

Reply:
Great find of the much simpler function @klonos - Done